test/labels.js

```diff
@@ -0,0 +1,349 @@
+                    return true;
+                }
+            }),
+
+            // ================
+            //     MID WAY
+            // ================
+            // now that we have established which pair was chosen in the first
+            // round we are going to do the same on a higher incarnation number
+            // in the other order.
+
+            // first send the second pair
+            dsl.changeStatus(t, tc, 0, 1, {
+                subjectIncNoDelta: +1,
+                status: 'alive',
+                labels: labels2,
+            }),
+            dsl.waitForPingResponse(t, tc, 0, 0, true),
+
+            // make sure these labels are accepted
+            dsl.assertMembership(t, tc, {
+                1: {
+                    labels: labels2
+                }
+            }),
+
+            // sending the first pair
+            dsl.changeStatus(t, tc, 0, 1, {
+                subjectIncNoDelta: 0,
+                status: 'alive',
+                labels: labels1,
+            }),
+            dsl.waitForPingResponse(t, tc, 0, 0, true),
+
+            // validate that the second iteration picked the same winning labels
+            // as the first iteration to ensure convergence when there are
+            // conflicting labels being gossipped around.
+            dsl.assertMembership(t, tc, {
+                1: {
+                    labels: winningLabels
+                }
+            }),
+        ];
+    })
+);
+
+test2('different labels should be accepted on a higher incarnation number', getClusterSizes(2), 20000,
+    prepareCluster(function(t, tc, n) {
+
+        // two pairs of labels
+        var labels1 = { "hello": "world" };
+        var labels2 = { "hello": "goodbey" };
+
+        return [
+            // feed the sut with the first pair of labels
+            dsl.changeStatus(t, tc, 0, 1, {
+                subjectIncNoDelta: +1,
+                status: 'alive',
+                labels: labels1,
+            }),
+            dsl.waitForPingResponse(t, tc, 0, 0, true),
+
+            // assert that the first pair is accepted
+            dsl.assertMembership(t, tc, {
+                1: { labels: labels1 }
+            }),
+
+             // feed the sut with the second pair of labels
+            dsl.changeStatus(t, tc, 0, 1, {
+                subjectIncNoDelta: +2,
+                status: 'alive',
+                labels: labels2,
+            }),
+            dsl.waitForPingResponse(t, tc, 0, 0, true),
+
+            // assert that the second pair is accepted
+            dsl.assertMembership(t, tc, {
+                1: { labels: labels2 }
+            }),
+        ];
+    })
+);
+
+// since nodes might miss updates on changes they should always accept labels if
+// the status updates, even if these labels differ from the labels that were
+// previously known to the node.
+function testLabelOverrideOnStatusChange(firstStatus, secondStatus) {
```

Comment:
I agree, for readability the invocations will be first and the definition later.